protos/sigstore_verification.proto

```diff
@@ -72,6 +76,14 @@ message ArtifactVerificationOptions {
                 // Disable signed timestamp verification.
                 bool disable = 2;
         }
+        message ObserverTimestampOptions {
```

Comment:
I can ask this on sigstore-go too, but why was "Observer" chosen, over "Trusted" or "Verified" or just "TimestampOptions"?

Reply:
I just went with observer as that's what's used in https://github.com/sigstore/sigstore-go/issues/44

I have no real preference, or maybe I do prefer `TimestampOptions` as long as it's clear compared to `TimestampAuthorityOptions` and that makes me thing that `VerifiedTimestampOptions` might be the best.

I kept the comment a bit longer on purpose to include my reasoning, hope it makes sense.